besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/TransactionEventPostProcessingSelector.java

```diff
@@ -0,0 +1,35 @@
+package net.consensys.linea.sequencer.txselection.selectors;
+
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import lombok.RequiredArgsConstructor;
+import net.consensys.linea.bundles.TransactionBundle;
+import org.hyperledger.besu.evm.log.Log;
+import org.hyperledger.besu.plugin.data.TransactionProcessingResult;
+import org.hyperledger.besu.plugin.data.TransactionSelectionResult;
+import org.hyperledger.besu.plugin.services.txselection.TransactionEvaluationContext;
+
+@RequiredArgsConstructor
+public class TransactionEventPostProcessingSelector
+    extends AbstractTransactionPostProcessingSelector {
+  private final AtomicReference<Set<TransactionEventSelectionDescription>> deniedEvents;
+  private final AtomicReference<Set<TransactionEventSelectionDescription>> deniedBundleEvents;
+
+  @Override
+  public TransactionSelectionResult evaluateTransactionPostProcessing(
+      final TransactionEvaluationContext evaluationContext,
+      final TransactionProcessingResult processingResult) {
+    final boolean isBundle =
+        evaluationContext.getPendingTransaction() instanceof TransactionBundle.PendingBundleTx;
+    Set<TransactionEventSelectionDescription> deniedEventsForTransaction =
+        isBundle ? deniedBundleEvents.get() : deniedEvents.get();
+    for (TransactionEventSelectionDescription deniedEvent : deniedEventsForTransaction) {
+      for (Log log : processingResult.getLogs()) {
+        if (deniedEvent.matches(log)) {
+          return TransactionSelectionResult.invalid("Transaction event logs match deny list");
```

Comment:
The pool validator is more descriptive about the rejection reason. I am wondering if we want to be more descriptive here as well:

```
  String.format(
      "recipient %s is blocked as appearing on the SDN or other legally prohibited list",
      to);
```